Deny popups

leong96 opened the pull request:
This completes the modal functionality for sending emails to the user for various rejections.

admin.html: Linked forms to the back end.
edit_denied_confirm_mail.txt: message altered to include the denied reason.
provider_account_denied_mail.txt: message altered to include the denied reason.
views.py: Re-imported some classes that were removed previously but are still used in views. Implemented code in the deny functions and the admin view to pass the reason to the context of the email.

leong96 committed:
Complete pop-ups

This completes the modal functionality for sending messages to the user.

admin.html: Linked forms to the back end.
edit_denied_confirm_mail.txt: message altered to include the denied reason.
provider_account_denied_mail.txt: message altered to include the denied reason.
views.py: Re-imported some classes that were removed previously but are still used in views. Implemented code in the deny functions and the admin view to pass the reason to the context of the email.

Removed comment regarding needing to insert a wrong password message. It's already there :D

justwheel commented on grasa_event_locator/views.py:
```diff
@@ -504,7 +509,7 @@ def register(request):
         i = 0
         if request.method == 'POST':
-                if User.objects.filter(username=request.POST['emailAddr']).exists():
+                if UserAccount.objects.filter(username=request.POST['emailAddr']).exists():
```
Just curious, why did these queries need to change from `User` -> `UserAccount`?

leong96 replied:
Regarding UserAccount, someone changed the import statement to import User as UserAccount. As such there were instances of "User" and "UserAccount" in the code. Any case of the above was switched to UserAccount for consistency (and to get things working again).

justwheel replied:
Oh huh, I didn't even notice. Nice catch!

justwheel committed:
templates: Add denied_reason to edit denied/provider denied emails

This commit adds the reason for denial to the event edit rejection email
and the provider account rejected emails. It builds on top of @leong96's
work.

Signed-off-by: Justin W. Flory <git@jwf.io>

justwheel reviewed (changes requested):
Hey @leong96, the code looks good to me although I had one question I left earlier.

I tried testing these changes manually, and I was not able to reject a new user registration or event edits. When I clicked the "Deny" button, I got a similar [stacktrace](http://dpaste.com/176MQTK) for both. It looks like the `denyUser()` method is executed before the front-end prompts the user for a rejection reason.

Rejecting new events worked for me.

Could you please take another look at this one?

leong96 committed:
Modal Return

At some point, the deny button links reverted to their old functionality (going directly  to deny_edit/deny_user). They have been re-hooked up to the modal.

justwheel reviewed (approved):
I tested this and it all works! Sorry if I muddled this while rebasing. Nice work @leong96. :100: 

Merging! :ocean:
